sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/HttpUrlConnectionAsyncClient.java

```diff
@@ -0,0 +1,206 @@
+                    for (String value : header.getValues()) {
+                        connection.addRequestProperty(header.getName(), value);
+                    }
+                }
+                if (httpRequest.getHttpMethod() != HttpMethod.GET) {
+                    connection.setDoOutput(true);
+                }
+
+                // Write body
+                switch (httpRequest.getHttpMethod()) {
+                    case POST:
+                    case PUT:
+                    case DELETE:
+                        if (httpRequest.getBody() != null) {
+                            try (DataOutputStream os = new DataOutputStream(new BufferedOutputStream(connection.getOutputStream()))) {
+
+                                Flux<ByteBuffer> requestBody = httpRequest.getBody();
+                                if (progressReporter != null) {
+                                    requestBody = requestBody.map(buffer -> {
+                                        progressReporter.reportProgress(buffer.remaining());
+                                        return buffer;
+                                    });
+                                }
+
+                                requestBody
+                                    .flatMap(buffer -> {
+                                        try {
+                                            byte[] bytes = new byte[buffer.remaining()];
+                                            buffer.get(bytes);
+                                            os.write(bytes);
+                                            return Mono.just(buffer); // Emit the buffer for downstream processing if needed
+                                        } catch (IOException e) {
+                                            return FluxUtil.monoError(LOGGER, new RuntimeException(e));
+                                        }
+                                    })
+                                    .then()
+                                    .block(); // Wait for completion of the write operations
+                                os.flush();
+                            } catch (IOException e) {
+                                return FluxUtil.monoError(LOGGER, new RuntimeException(e));
+                            }
+                        }
+                        break;
+                    case GET:
+                    case HEAD:
+                    case OPTIONS:
+                    case TRACE:
+                    case CONNECT:
+                        break;
+                    default:
+                        return FluxUtil.monoError(LOGGER, new IllegalStateException("Unknown HTTP Method:"
+                            + httpRequest.getHttpMethod()));
+                }
+
+                // Read response
+                int responseCode = connection.getResponseCode();
+
+                Map<String, List<String>> responseHeadersMap = new HashMap<>();
```

Comment:
Since `HttpResponse` exposes `HttpHeader`/`HttpHeaders` APIs I would use `HttpHeaders` here instead of the map. This will remove the need to map later on.